apiserver/facades/controller/caasunitprovisioner/provisioner_test.go

```diff
@@ -351,7 +354,7 @@ func (s *CAASProvisionerSuite) TestUpdateApplicationsStatelessUnits(c *gc.C) {
 	}
 	args := params.UpdateApplicationUnitArgs{
 		Args: []params.UpdateApplicationUnits{
-			{ApplicationTag: "application-gitlab", Units: units, Scale: intPtr(4)},
+			{ApplicationTag: "application-gitlab", Units: units, Scale: intPtr(4), Generation: int64Ptr(1)},
```

Comment:
also need a test for nil generation for older operators

Reply:
added a test